web3.js/src/util/send-and-confirm-transaction.ts

```diff
@@ -27,23 +27,47 @@ export async function sendAndConfirmTransaction(
     maxRetries: options.maxRetries,
   };
 
+  let status;
+
   const signature = await connection.sendTransaction(
     transaction,
     signers,
     sendOptions,
   );
 
-  const status = (
+  status =
+    transaction.recentBlockhash != null &&
+    transaction.lastValidBlockHeight != null
+      ? (
+          await connection.confirmTransaction(
+            {
+              signature: signature,
+              blockhash: transaction.recentBlockhash!,
+              lastValidBlockHeight: transaction.lastValidBlockHeight!,
```

Comment:
Since you asserted that these are non-null 6 lines above here, you don't need to add the trailing `!` anymore. Remove it, and make sure that TypeScript knows for sure they exist.

Reply:
Done!